Penteract/Assets/Scripts/DialogueManager.cpp

```diff
@@ -0,0 +1,308 @@
+#include "DialogueManager.h"
+#include "DialogueManager.h"
+
+#include "GameplaySystems.h"
+#include "GameObject.h"
+#include "Components/UI/ComponentText.h"
+
+// TODO:
+// Collisions with everything?
+// F outside dialoges should do nothing
+// IsBlocking to "pause" the gameplay
+// Transition effect between dialogue boxes
+
+EXPOSE_MEMBERS(DialogueManager) {
+	MEMBER(MemberType::GAME_OBJECT_UID, fangTextObjectUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, onimaruTextObjectUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, tutorialFangUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, tutorialOnimaruUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, tutorialSwapUID),
+	MEMBER(MemberType::GAME_OBJECT_UID, tutorialUpgrades1UID),
+	MEMBER(MemberType::GAME_OBJECT_UID, tutorialUpgrades2UID),
+	MEMBER(MemberType::GAME_OBJECT_UID, tutorialUpgrades3UID),
+	MEMBER(MemberType::FLOAT3, dialogueStartPosition),
+	MEMBER(MemberType::FLOAT3, dialogueEndPosition),
+	MEMBER(MemberType::FLOAT3, tutorialStartPosition),
+	MEMBER(MemberType::FLOAT3, tutorialEndPosition),
+	MEMBER(MemberType::FLOAT, appearAnimationTime),
+	MEMBER(MemberType::FLOAT, disappearAnimationTime)
+};
+
+GENERATE_BODY_IMPL(DialogueManager);
+
+void DialogueManager::Start() {
+	GameObject* fangTextObject = GameplaySystems::GetGameObject(fangTextObjectUID);
+	GameObject* onimaruTextObject = GameplaySystems::GetGameObject(onimaruTextObjectUID);
+	tutorialFang = GameplaySystems::GetGameObject(tutorialFangUID);
+	tutorialOnimaru = GameplaySystems::GetGameObject(tutorialOnimaruUID);
+	tutorialSwap = GameplaySystems::GetGameObject(tutorialSwapUID);
+	tutorialUpgrades1 = GameplaySystems::GetGameObject(tutorialUpgrades1UID);
+	tutorialUpgrades2 = GameplaySystems::GetGameObject(tutorialUpgrades2UID);
+	tutorialUpgrades3 = GameplaySystems::GetGameObject(tutorialUpgrades3UID);
+
+	if (fangTextObject && onimaruTextObject) {
+		fangTextComponent = fangTextObject->GetComponent<ComponentText>();
+		onimaruTextComponent = onimaruTextObject->GetComponent<ComponentText>();
+	}
+
+	// UPGRADES
+	dialoguesArray[0] = Dialogue(2, "Hey Fang, look at this.", &dialoguesArray[1]);
```

Comment:
I would use an Enum for Interlocutors

Reply:
I thought about it, but i wanted to finish the functionality asap.
I'm still missing some secondary features (as stated in TODO), I can add it also to the TODO list and work on it too for the next Alpha

Reply:
@FrancPS Yeah, that's fine. That can be changed later on, NP.